homeassistant/components/livisi/switch.py

```diff
@@ -95,74 +99,56 @@ def __init__(
         self._attr_name = name
         self._state = None
         self._capability_id = capability_id
-        self._attr_is_on = is_on
         self._room = room
         self._version = version
         self.aio_livisi = coordinator.aiolivisi
-        if is_on is None:
-            self._attr_is_available = False
-        else:
-            self._attr_is_available = True
+        self._attr_is_available = False
         self._attr_device_info = DeviceInfo(
             identifiers={(DOMAIN, str(self._attr_unique_id))},
             manufacturer=self._manufacturer,
             model=self._attr_model,
             name=self._attr_name,
             suggested_area=self._room,
             sw_version=self._version,
-            via_device=(DOMAIN, str(self._attr_unique_id)),
         )
         super().__init__(coordinator)
 
-    @property
-    def should_poll(self) -> bool:
-        """No polling needed."""
-        return False
-
-    @property
-    def is_on(self) -> bool | None:
-        """Return the device state."""
-        return self._attr_is_on
-
-    @property
-    def available(self) -> bool:
-        """Return if switch is available."""
-        return self._attr_is_available
-
     async def async_turn_on(self, **kwargs: Any) -> None:
         """Turn the entity on."""
-        if self.is_on is True:
-            return
         response = await self.aio_livisi.async_pss_set_state(
             self._capability_id, is_on=True
         )
         if response is None:
             self._attr_is_available = False
+            raise HomeAssistantError(f"Failed to turn on {self._attr_name}")
 
     async def async_turn_off(self, **kwargs: Any) -> None:
         """Turn the entity off."""
-        if self.is_on is False:
-            return
         response = await self.aio_livisi.async_pss_set_state(
             self._capability_id, is_on=False
         )
         if response is None:
             self._attr_is_available = False
+            raise HomeAssistantError(f"Failed to turn off {self._attr_name}")
 
     async def async_added_to_hass(self) -> None:
         """Register callbacks."""
-        self.coordinator.async_add_listener(self.update_states)
-        self.coordinator.async_add_listener(self.update_reachability)
+        self._attr_is_on = await self.coordinator.async_get_pss_state(
+            self._capability_id
+        )
+        async_dispatcher_connect(self.hass, LIVISI_STATE_CHANGE, self.update_states)
```

Comment:
Use `self.async_on_remove` to unsubscribe the dispatch listener on entity remove.

Reply:
I added this, thank you.